pandas/core/groupby/ops.py

```diff
@@ -229,7 +231,7 @@ def levels(self) -> List[Index]:
         return [ping.group_index for ping in self.groupings]
 
     @property
-    def names(self):
+    def names(self) -> List[str]:
```

Comment:
are these necessarily strings, or are they Labels?

Reply:
I'm not positive they have to be a strings so Labels seems safer

Reply:
it looks like each entry here corresponds to the .name attr of a grouper.Grouping object.  That isnt annotated (and the docstring entry for name doesnt have a type).  Can you track down where that attr comes from to be sure, and if so, add the annotation there too.

Reply:
It seems this is just the "key" of the thing we group by, so "a" in df.groupby("a"), or 1 in df.groupby(1). It can also be None (as in a same-length grouper) so Label probably makes sense since it's Optional[Hashable]